### PR TITLE
VAULT-8790 Ensure time.NewTicker never gets called with a negative value

### PR DESCRIPTION
**Pull request by VioletHynes**
`time.NewTicker` panics if given a value `<= 0`. As a result, it needs to be given a positive number (above 0). These were all of the cases I found where this could happen, and I added a test for the main one called out in the bug report.

### CODE REVIEW

File: command/agentproxyshared/auth/ldap/ldap.go [hunk 1]
```diff
@@ -107,8 +107,8 @@ func NewLdapAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 
 	if passReadPeriodRaw, ok := conf.Config["password_read_period"]; ok {
 		passReadPeriod, err := parseutil.ParseDurationSecond(passReadPeriodRaw)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing 'pass_read_period' value: %w", err)
+		if err != nil || passReadPeriod <= 0 {
+			return nil, fmt.Errorf("error parsing 'pass_read_period' value into a positive value: %w", err)
```
Comment (peteski22) [inline command/agentproxyshared/auth/ldap/ldap.go:111]:
```suggestion
			return nil, fmt.Errorf("error parsing 'password_read_period' value into a positive value: %w", err)
```

File: builtin/logical/database/rotation_test.go [hunk 2]
```diff
@@ -998,6 +999,34 @@ func TestBackend_StaticRole_Rotation_MongoDBAtlas(t *testing.T) {
 	})
 }
 
+// TestQueueTickIntervalKeyConfig tests the configuration of queueTickIntervalKey
+// Previous to a fix, this test panicked.
```
Comment (peteski22) [inline builtin/logical/database/rotation_test.go:1003]:
```suggestion
```

File: builtin/logical/database/rotation_test.go [hunk 3]
```diff
@@ -998,6 +999,34 @@ func TestBackend_StaticRole_Rotation_MongoDBAtlas(t *testing.T) {
...
+func TestQueueTickIntervalKeyConfig(t *testing.T) {
```
Comment (peteski22) [inline builtin/logical/database/rotation_test.go:1004]: I'm not sure I completely understand this, so apologies if this is a foolish one.

I was expecting to see some output in the logs when I ran this test locally to say it was using the default value for `0` and `-1` cases,

e.g. `bad value for...` from: https://github.com/hashicorp/vault/blob/05b37831c4e470ef9fa18f8444d97491204648fb/builtin/logical/database/rotation.go#L592

but it just says the context is cancelled.

image

Is that what's expected?

**Review by peteski22 — COMMENTED** [review]
Nice one Violet, I felt slightly terrified that there'd be tonnes of place that we `time.NewTicker` which would make life a pain. 

I added a comment about the test as I was confused. 😳

File: builtin/logical/database/rotation_test.go (hunk 3 above)
Comment (VioletHynes) [reply builtin/logical/database/rotation_test.go:1004]: I wrote this test before my fix, and what happened for me is the test failed with a panic

File: command/agentproxyshared/auth/ldap/ldap.go (hunk 1 above)
Comment (VioletHynes) [reply command/agentproxyshared/auth/ldap/ldap.go:111]: Hah, good catch! Thanks :)

**Review by marcboudreau — APPROVED** [review]

**Pull request merged**